backend/compact-connect/README.md

```diff
@@ -65,56 +65,66 @@ To add additional dependencies, for example other CDK libraries, just add them t
 [Back to top](#compact-connect---backend-developer-documentation)
```

Comment:
> On my ToDo list is standing up the site locally using these updated instructions. I'll provide feedback on the instructions separately. Related to this, would the IA team prefer I provide feedback in waves or just once when I've completed my review from end-to-end?

With this level of depth (thank you), I'm totally on board with you doing waves, since I can start replying/revising without waiting for you to finish the whole thing.

> I'd like to see longer and more descriptive names for tests.

Sure thing.

> There are a few places where code is repeated more than twice (my threshold generally for abstracting), in tests and sometimes elsewhere. It'd be great to keep this DRY.

I can do that. I've been leaning into a style of writing tests in a very linear, self-contained (but very much not DRY) approach, lately but don't mind changing if it bothers you.

> Going forward, can we do PRs by user story rather than a generic 'feature'? Under that model, this PR would presumably also include frontend code.

That absolutely makes sense. I think we'll be moving in that direction as the project gets further into user-features and less into the foundational work. It's probably worth a conversation with @isabeleliassen so we're all on the same page about how the work is being planned.

> Nit: There are a few unnecessary pylint suppressions listed in the CI/CD output.

Yep. Those are there for a reason. Pylint has been somewhat inconsistent about some of its findings for reasons that I haven't gotten to the bottom of. I think it has to do with the order that it analyzes files - if it sees a name used in one module, but that name is defined in a module it hasn't analyzed yet, it seems to skip on checks that require it to know more about the name. Because of that, it is inconsistent about when it catches those findings. I'm just leaving them suppressed, even if it then flags the suppressions as unnecessary. Pylint doesn't deduct from its linting score for unnecessary suppressions, but does for the findings if they aren't suppressed. An unnecessary suppression adds noise to the report, but an un-suppressed finding will fail the PR.

> For a future sprint: could we automate an audit on python dependencies (perhaps with pip-lint) and incorporate a static code analysis tool like SonarCloud? I can run these ad-hoc but it'll be more and more of a pain with feature-based PRs.

Sure. I'm doing this as a pre-commit check, currently. I can add it to the github automation alongside linting and testing.

Reply:
Re: DRY- your philosophy totally makes sense. As long as it's intentional, I'm fine with taking that approach in test code.

Re: user stories vs features, agreed that we should all sync to get on the same page!